File: solutions/csharp/01-ry8/code/src/main.cs
using System;
using System.IO;

if (args.Length < 2)
{
    Console.Error.WriteLine("Usage: ./your_program.sh tokenize <filename>");
    Environment.Exit(1);
}

string command = args[0];
string filename = args[1];

if (command != "tokenize")
{
    Console.Error.WriteLine($"Unknown command: {command}");
    Environment.Exit(1);
}

string fileContents = File.ReadAllText(filename);

if (!string.IsNullOrEmpty(fileContents))
{
    throw new NotImplementedException("Scanner not implemented");
}
else
{
    Console.WriteLine("EOF  null"); // Placeholder, replace this line when implementing the scanner
}

File: solutions/haskell/01-ry8/code/app/Main.hs
{-# OPTIONS_GHC -Wno-unused-top-binds #-}
{-# LANGUAGE BlockArguments #-}
{-# LANGUAGE OverloadedStrings #-}

module Main (main) where

import System.Environment (getArgs)
import System.Exit (exitWith, exitFailure, ExitCode(ExitSuccess))
import System.IO (hPutStrLn, hSetBuffering, stdout, stderr, BufferMode(NoBuffering), readFile)

main :: IO ()
main = do
    -- Disable output buffering
    hSetBuffering stdout NoBuffering
    hSetBuffering stderr NoBuffering

    args <- getArgs
    case args of
        (command : filename : _) ->
            if command /= "tokenize"
                then do
                    hPutStrLn stderr $ "Unknown command: " ++ command
                    exitFailure
                else do
                    fileContents <- readFile filename
                    if not (null fileContents)
                        then error "Scanner not implemented"
                        else putStrLn "EOF  null"  -- Placeholder, replace this line when implementing the scanner
                    pure ()
        _ -> do
            hPutStrLn stderr "Usage: ./your_program.sh tokenize <filename>"
            exitFailure
